CDCgov/prime-reportstream
# Log the element type and UUID for FHIR elements filtered out by Condition Filters

This PR updates the logging of filter results on submitted reports. When filters are run on specific parts of the report's Bundle, the type and UUID for the filtered element is included on the log.

Note - This PR has attached a CURL call that I used on Postman to test. Feel free to use it or make your own test.

Test Steps:
1. In organizations.yml, update the `Ignore FULL_ELR` receiver to have a conditionFilter that will cause at least one `OBX` segment to be filtered out. e.g. `conditionFilter: ["%obsPerformedCodes.intersect('123-4').exists()"]`
2. Run ReportStream, then run `./gradlew reloadSettings` to update the settings.
3. Submit a report with data that would go to the `Ignore FULL_ELR_FHIR` Receiver.
4. Using the History API, verify that at least one row is inside `filteredReportRows` for `Ignore FULL_ELR_FHIR` in the `destinations` list.
5. Verify that the output on the Report Rows matches the format: `For ignore.FULL_ELR_FHIR, filter [FILTERDETAILS] filtered out item [ITEMID] at Observation/[UUID]"`, with relevant values replacing the bracketed contents in the example.

## Changes
- Update `FHIRRouter` to include the part of the item that's been filtered out.

## Linked Issues
- #8689  

[local call curl.txt](https://github.com/CDCgov/prime-reportstream/files/11534718/local.call.curl.txt)
